azure-spring-maven-plugin/src/main/java/com/microsoft/azure/maven/spring/prompt/DefaultPrompter.java

```diff
@@ -0,0 +1,287 @@
+            throw new IllegalArgumentException("There is no way to input empty value for a non-required field with default value.");
+        }
+        final String defaultValueMessage = hasDefaultValue ? " (" + TextUtils.blue(defaultValue.toString()) + ")" : "";
+        final String hintMessage = String.format("[%d-%d]%s", minValue, maxValue, defaultValueMessage);
+        final String message = String.format("Please input the %s %s: ", name, hintMessage);
+        System.out.print(message);
+        System.out.flush();
+
+        return loopInput(defaultValue, hasDefaultValue, isRequired, message, input -> {
+            return validateUserInputAsInteger(input, minValue, maxValue, String.format(PROMPT_MESSAGE_NUM_TEMPLATE, input, name, hintMessage));
+        });
+    }
+
+    static class UserInputException extends RuntimeException {
+        private static final long serialVersionUID = 7986173597822766589L;
+
+        UserInputException(String message) {
+            super(message);
+        }
+    }
+
+    private <T> T loopInput(T defaultValue, boolean hasDefaultValue, boolean isRequired, String emptyPromoteMessage,
+            Function<String, T> handleInput) throws IOException {
+        while (true) {
+            final String input = reader.readLine();
+            if (StringUtils.isBlank(input)) {
+                if (hasDefaultValue || !isRequired) {
+                    return defaultValue;
+                }
+                System.out.print(emptyPromoteMessage);
+            } else {
+                try {
+                    return handleInput.apply(input);
+                } catch (UserInputException ex) {
+                    System.out.print(ex.getMessage());
+                }
+            }
+            System.out.flush();
+        }
+    }
+
+    private Integer validateUserInputAsInteger(String input, int start, int end, String message) {
+        if (!NumberUtils.isDigits(input)) {
+            throw new UserInputException(message);
+        }
+
+        final int value = Integer.parseInt(input);
+        if (value >= start && value <= end) {
+            return value;
+        }
+
+        throw new UserInputException(message);
+    }
+
+    public void close() {
+        try {
+            reader.close();
+        } catch (IOException e) {
+            // swallow this error
+        }
+    }
+
+    private static String yesOrNo(Boolean defaultValue) {
+        if (defaultValue == null) {
+            return "(y/n)";
+        }
+        if (defaultValue.booleanValue()) {
+            return "(Y/n)";
+        }
+        return "(y/N)";
+    }
+
+    private static boolean isValidIntRangeInput(String text) {
+        return REGEX_COMMA_SEPARATED_INTEGER_RANGES.matcher(text).matches();
+    }
+
+    private static int[] parseIntRanges(String text, int minValue, int maxValue) {
+        final Matcher m = REGEX_NEXT_INTEGER_RANGE.matcher(text);
+        final List<Integer> values = new ArrayList<>();
```

Comment:
Use a HashSet here, so that no need to remove duplicated items with stream later

Reply:
done without hashset since it doesn't keep order.